ec/ecresource/deploymentresource/elasticsearch_flatteners.go

```diff
@@ -71,6 +71,10 @@ func flattenEsResources(in []*models.ElasticsearchResourceInfo, name string, rem
 			m["config"] = c
 		}
 
+		if s := flattenSnapshotSource(plan); len(s) > 0 {
```

Comment:
Since the restore snapshot is supposed to be a one-off (since it lives in the transient block part of the plan) operation I think that it makes sense to omit the flattening so that any time that the `elasticsearch.#.snapshot_source` is set, the user gets a diff with additions on that part.

This would also require documenting the behavior but it would have less disruptive consequences than flattening and re-applying that state at a later point in time.

Reply:
Yeah, I was on the fence about adding the flattening behaviour 🤔. Given it already confused someone, I think you're right and it's best to remove it.